sunxspex/sun_xspec.py

```diff
@@ -0,0 +1,349 @@
+            photon fluxes in photons s^-1 keV^-1 cm^-2, corresponding to the photon energies in the
+            input array eph. The detector is assumed to be 1 AU rom the source. The coefficient a0 is
+            calculated as a0 = nth * V * nnth, where nth: plasma density; cm^-3) V:
+            volume of source; cm^3) nnth: Integrated nonthermal electron flux density (cm^-2 s^-1), if
+            efd = True, or Integrated electron number density (cm^-3), if efd = False
+
+        Notes
+        -----
+        If you want to plot the derivative of the flux, or the spectral index of the photon spectrum as
+        a function of photon energy, you should set RERR to 1.d-6, because it is more sensitive to RERR
+        than the flux.
+
+        Adapted from SSW `Brm2_ThinTarget
+        <https://hesperia.gsfc.nasa.gov/ssw/packages/xray/idl/brm2/brm2_thintarget.pro>`_
+        """
+        mc2 = const.get_constant('mc2')
+        clight = const.get_constant('clight')
+        au = const.get_constant('au')
+
+        # Max number of points
+        maxfcn = 2048
+
+        # Average atomic number
+        z = 1.2
+
+        # Relative error
+        rerr = 1e-4
+
+        # Numerical coefficient for photo flux
+        fcoeff = (clight / (4 * np.pi * au ** 2)) / mc2 ** 2.
+
+        # Create arrays for the photon flux and error flags.
+        flux = np.zeros_like(photon_energies, dtype=np.float64)
+        iergq = np.zeros_like(photon_energies, dtype=np.float64)
+
+        if eelow >= eehigh:
+            raise ValueError('eehigh must be larger than eelow!')
+
+        l, = np.where((photon_energies < eehigh) & (photon_energies > 0))
+        if l.size > 0:
+            flux[l], iergq[l] = split_and_integrate(model='thin-target',
+                                                    photon_energies=photon_energies[l], maxfcn=maxfcn,
+                                                    rerr=rerr, eelow=eelow, eebrk=eebrk, eehigh=eehigh,
+                                                    p=p, q=q, z=z, efd=efd)
+
+            flux *= fcoeff
+
+            return flux
+        else:
+            raise Warning('The photon energies are higher than the highest electron energy or not '
+                          'greater than zero')
+
+
+class ThermalModel(XspecModel):
+    def __init__(self):
+        self.ParInfo=(
+        "EM  1e49  1.0  1e-20  1e-19  1e19  1e20  1.0",
+        "kT  keV  2.0  1.  1.5  7.0  8.0  0.1",
+        "abund  \"\"  1.0  0.01  0.1  9.0  10.0  0.1") #default parameters from OSPEX
+        self.model=self.vth
+        self.description=f"Thermal bremsstrahlung model '{self.model.__name__}'"
+        global CONTINUUM_GRID
+        CONTINUUM_GRID = thermal.setup_continuum_parameters()
+        global LINE_GRID
+        LINE_GRID= thermal.setup_line_parameters()
+        #self.observer_distance=(1*u.AU).to(u.cm).value
+
+    @staticmethod
+    def vth(energy_edges,params,flux):
```

Comment:
Same question here regarding why this doesn't call the thermal function.

Also, same point about using self-explanatory names rather than ambiguous IDL ones.